### ISSUE TITLE
The training loss on tensorboard is half of the one on progbar when accumulate_grad_batches=2

### ISSUE THREAD

**Issue by quancs** [Lightning-AI/lightning#9767]
## 🐛 Bug

The training loss on tensorboard is half of the one on progbar when accumulate_grad_batches=2

I guess it's because training loss is divided by `accumulate_grad_batches` instead of average over `accumulate_grad_batches` batches, but I'm not sure

### Expected behavior

The training loss on tensorboard should equal to the one on progbar

### Environment

- PyTorch Lightning Version (e.g., 1.3.0): 1.5.0dev
- PyTorch Version (e.g., 1.8)
- Python version:
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

**Comment by awaelchli** [comment]
Hey @quancs 
I will double check your observation. But we only log to the logger on optimizer steps (so at the end of accumulation phases) while the progress bar shows every update (although a smoothed value over a window of a few steps).

If I understand correctly: 
```
self.log("train_loss", loss, logger=True, prog_bar=True)
return {"loss": loss}
```
Will do two things:
Logger will show the values in a plot "train_loss" as they are logged there on every optimizer step, no modifications done. 
Progress bar will show: a) "train_loss" as logged by you but updated not only on optimizer steps but each backward b) will show "loss" what you returned from training_step and there compute the running average over the accumulation phase.

**Comment by quancs** [comment]
@awaelchli 
For my case, the training loss on tensorboard is approximately 6, the training loss on prog bar is approximately 12, the validation loss on tensorboard is approximately 12, the test loss is approximately 12. (PL version 1.5.0dev)
Hope this will help

**Comment by quancs** [comment]
@awaelchli When I use two GPUs training, the loss on progbar becomes as twice as the metric on tensorboard (accumulate_grad_batches=1, sync_dist=True) and the training loss on tensorboard is at the same level with val loss on tensorboard now...

This is very annoying. Can you give me two breakpoints for observing the metrics logged on tensorboard and on progbar (the pytorch_lightning is big, I'm not sure where are they)

**Comment by awaelchli** [comment]
This does no longer happen after Lightning 2.0, because we no longer automatically log the loss to the progress bar. The user would log it themselves with `self.log` and so the exact value is shown in the bar. This means also the loss value in the logger will be the exact same. Closing the issue.
